### PR TITLE
Migrate tests to JUnit5

### PR DESCRIPTION
**Pull request by Tristan-WorkGH**
Need powsybl/powsybl-parent#65

### CODE REVIEW

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]
That's a lot of unnecessary changes, please revert them. And you created a lot of sonar issues by using a deprecated constant.

File: src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java [hunk 1]
```diff
@@ -118,32 +100,20 @@ public MockResponse dispatch(RecordedRequest request) {
...
                 } else if (path.matches("/v1/directories/.*/elements\\?allowNewName=.*") && "POST".equals(request.getMethod())) {
                     ElementAttributes elementAttributes = new ElementAttributes(CONFIG_UUID, CONFIG_NAME, "SPREADSHEET_CONFIG", USER_ID, 0L, null);
-                    return new MockResponse()
-                            .setResponseCode(200)
-                            .setHeader("Content-Type", "application/json")
-                            .setBody(objectMapper.writeValueAsString(elementAttributes));
+                    return new MockResponse(200, Headers.of("Content-Type", "application/json"), objectMapper.writeValueAsString(elementAttributes));
                 } else if (path.matches("/v1/directories/" + PARENT_DIRECTORY_UUID + "/elements")) {
-                    return new MockResponse().setResponseCode(200).addHeader("Content-Type", "application/json; charset=utf-8");
+                    return new MockResponse(200, Headers.of("Content-Type", "application/json; charset=utf-8") /*, body: TODO content ???*/);
```
Comment (TheMaskedTurtle) [inline src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java:108]: In other services you just put MockResponse(200)

File: src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java [hunk 2]
```diff
@@ -61,53 +65,31 @@ class SpreadsheetConfigTest {
...
         directoryService.setDirectoryServerBaseUri(baseUrl);
 
         spreadsheetConfigJson = "{\"sheetType\":\"GENERATORS\",\"customColumns\":[{\"id\":\"" + UUID.randomUUID() + "\",\"name\":\"Custom Column\",\"formula\":\"SUM(A1:A10)\"}]}";
 
         mockWebServer.setDispatcher(new Dispatcher() {
+            @NotNull
             @SneakyThrows
             @Override
-            public MockResponse dispatch(RecordedRequest request) {
+            public MockResponse dispatch(@NotNull RecordedRequest request) {
```
Comment (TheMaskedTurtle) [inline src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java:80]: Do we really need all the @NotNull ?

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java [hunk 3]
```diff
@@ -456,141 +413,141 @@ public void testCaseCreationError() throws Exception {
...
 
     @Test
-    public void testCreateScriptContingencyList() throws Exception {
+    void testCreateScriptContingencyList() throws Exception {
         mockMvc.perform(post("/v1/explore/script-contingency-lists/{listName}?&parentDirectoryUuid={parentDirectoryUuid}&description={description}}",
                 "contingencyListScriptName", PARENT_DIRECTORY_UUID, null)
                 .header("userId", USER1)
                 .contentType(MediaType.APPLICATION_JSON)
-                .content("Contingency list content")
+                .content("\"Contingency list content\"")
```
Comment (TheMaskedTurtle) [inline src/test/java/org/gridsuite/explore/server/ExploreTest.java:421]: Why do you need to change that ?

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java [hunk 4]
```diff
@@ -166,251 +164,210 @@ public void setup() throws IOException {
...
                 } else if (path.matches("/v1/parameters.*")) {
-                    return new MockResponse().setResponseCode(200);
+                    return new MockResponse(200);
                 } else if (path.matches("/v1/network-composite-modifications")) {
-                    return new MockResponse().setBody(compositeModificationIdAsString).setResponseCode(200).addHeader("Content-Type", "application/json; charset=utf-8");
+                    return new MockResponse(200, Headers.of(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE), compositeModificationIdAsString);
                 } else if (path.matches("/v1/messages/" + USER_WITH_CASE_LIMIT_NOT_EXCEEDED + "/user-message.*")) {
-                    return new MockResponse().setResponseCode(200)
-                        .addHeader("Content-Type", "application/json; charset=utf-8");
+                    return new MockResponse(200, Headers.of("Content-Type", "application/json; charset=utf-8") /*, body: TODO content??*/);
```
Comment (TheMaskedTurtle) [inline src/test/java/org/gridsuite/explore/server/ExploreTest.java:279]: Same just put MockResponse(200)

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java (hunk 2 above)
Comment (Tristan-WorkGH) [reply src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java:80]: No.

File: src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java (hunk 1 above)
Comment (Tristan-WorkGH) [reply src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java:108]: Yes, replace-all haven't take that line after.

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java (hunk 3 above)
Comment (Tristan-WorkGH) [reply src/test/java/org/gridsuite/explore/server/ExploreTest.java:421]: The response mock return JSON, but we were returning just plain text...

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java (hunk 4 above)
Comment (TheMaskedTurtle) [reply src/test/java/org/gridsuite/explore/server/ExploreTest.java:279]: You didn't fix this one

File: src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java (hunk 2 above)
Comment (Tristan-WorkGH) [reply src/test/java/org/gridsuite/explore/server/SpreadsheetConfigTest.java]: yes

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java (hunk 4 above)
Comment (Tristan-WorkGH) [reply src/test/java/org/gridsuite/explore/server/ExploreTest.java:279]: oups

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**